presto-hive-metastore/src/main/java/com/facebook/presto/hive/HiveBucketProperty.java

```diff
@@ -34,16 +36,42 @@
     private final List<String> bucketedBy;
     private final int bucketCount;
     private final List<SortingColumn> sortedBy;
+    private final Optional<List<Type>> types;
+    private final BucketingType bucketingType;
+
+    public HiveBucketProperty(
+            List<String> bucketedBy,
+            int bucketCount,
+            List<SortingColumn> sortedBy)
+    {
+        this(bucketedBy, bucketCount, sortedBy, Optional.empty());
+    }
 
     @JsonCreator
     public HiveBucketProperty(
             @JsonProperty("bucketedBy") List<String> bucketedBy,
             @JsonProperty("bucketCount") int bucketCount,
-            @JsonProperty("sortedBy") List<SortingColumn> sortedBy)
+            @JsonProperty("sortedBy") List<SortingColumn> sortedBy,
+            @JsonProperty("types") Optional<List<Type>> types)
     {
         this.bucketedBy = ImmutableList.copyOf(requireNonNull(bucketedBy, "bucketedBy is null"));
         this.bucketCount = bucketCount;
         this.sortedBy = ImmutableList.copyOf(requireNonNull(sortedBy, "sortedBy is null"));
+        this.types = requireNonNull(types, "type is null");
+        if (types.isPresent()) {
+            bucketingType = BucketingType.PRESTOTYPE;
+            checkArgument(types.get().size() > 0, "types can not be empty if present");
+            checkArgument(types.get().size() == bucketedBy.size(), "the sizes of bucketedBy and types should match");
+        }
+        else {
+            bucketingType = BucketingType.HIVETYPE;
+        }
+    }
+
+    public enum BucketingType
+    {
+        HIVETYPE,
```

Comment:
`HIVE_COMPATIBLE`, `PRESTO_NATIVE`

Reply:
How about `HIVE_TYPE` and `PRESTO_NATIVE`?  `COMP
ATIBLE` is a little confused , as there is no translation here.